code/eyetracking/eyetrackingrun.py

```diff
@@ -0,0 +1,781 @@
+            recorded_eye,
+            eye_tracking_method,
+            sampling_frequency,
+            pupil_threshold,
+            CR_threshold,
+            pupil_fit_method,
+        ) = self.extract_ET_parameters()
+        header = self.extract_header()
+        final_info = {
+            "Manufacturer": info_ET["Manufacturer"],
+            "ManufacturersModelName": info_ET["ManufacturersModelName"],
+            "DeviceSerialNumber": info_ET["DeviceSerialNumber"],
+            "SoftwareVersion": info_ET["SoftwareVersion"],
+            "CalibrationUnit": info_ET["CalibrationUnit"],
+            "EyeTrackerDistance": info_ET["EyeTrackerDistance"],
+            "SampleCoordinateUnits": info_ET["SampleCoordinateUnits"],
+            "SampleCoordinateSystem": info_ET["SampleCoordinateSystem"],
+            "EnvironmentCoordinates": info_ET["EnvironmentCoordinates"],
+            "ScreenAOIDefinition": info_ET["ScreenAOIDefinition"],
+            "SamplingFrequency": sampling_frequency,
+            "StartTime": timestamp_first_trigger,
+            "RecordedEye": recorded_eye,
+            "EyeTrackingMethod": eye_tracking_method,
+            "PupilFitMethod": pupil_fit_method,
+            "GazeMappingSettings": {
+                "CRThreshold": CR_threshold,
+                "PThreshold": pupil_threshold,
+            },
+            "CalibrationCount": calibration_count,
+            "Columns": self.samples.columns.tolist(),
+            "CalibrationType": calibration_type,
+            "CalibrationPosition": calibration_position,
+            "AverageCalibrationError": average_calibration_error,
+            "MaximalCalibrationError": max_calibration_error,
+            "EDFHeader": header,
+        }
+
+        final_info = {
+            key: value for key, value in final_info.items() if value is not None
+        }
+
+        if not final_info:
+            print("No data to write to the JSON file. Skipping.")
+            return None
+
+        if self.task_name in ["rest", "bht", "qct"]:
+            output_file_name = f"sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_dir-{self.pe}_eyetrack.json"
+            output_file_dir = os.path.join(
+                BIDS_folder_path,
+                f"sub-{self.participant:03d}/ses-{self.session:03d}/func/",
+            )
+        elif self.task_name == "dwi":
+            output_file_name = f"sub-{self.participant:03d}_ses-{self.session:03d}_acq-highres_dir-{self.pe}_eyetrack.json"
+            output_file_dir = os.path.join(
+                BIDS_folder_path,
+                f"sub-{self.participant:03d}/ses-{self.session:03d}/dwi/",
+            )
+        else:
+            output_file_name = f"sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_eyetrack.json"
+            output_file_dir = os.path.join(
+                BIDS_folder_path,
+                f"sub-{self.participant:03d}/ses-{self.session:03d}/func/",
+            )
```

Comment:
You don't want to be here
```suggestion
            raise ValueError("Unknown task type")
```

Reply:
I need this part to process the data from other projects (e.g. Benedetta's data)

Reply:
why was the suggestion ignored?

Reply:
I didn't ignore the suggestion; this line was meant to keep the option open for processing other data, like Benedetta's. I'm ok with removing it, just wanted to confirm before making changes.